File: Dummy/hawkshaw/library/FileServiceGrpc.java
package com.hawkshaw.library;

import X1.B;
import Y1.A0;
import Y1.C0104e;
import Y1.C0106f;
import Y1.C0108g;
import Y1.C0114j;
import Y1.o0;
import Y1.p0;
import Y1.y0;
import Y1.z0;
import com.google.protobuf.C0385x;
import g2.C0452b;
import g2.C0453c;
import i.s0;
import io.grpc.stub.a;
import io.grpc.stub.b;
import io.grpc.stub.c;
import io.grpc.stub.f;
import io.grpc.stub.g;
import io.grpc.stub.i;
import io.grpc.stub.j;
import java.util.List;
import p3.q;

public final class FileServiceGrpc {
    private static final int METHODID_BATCH_THUMB_UPLOAD = 1;
    private static final int METHODID_UPLOAD_FILE = 0;
    public static final String SERVICE_NAME = "filecrud.FileService";
    private static volatile p0 getBatchThumbUploadMethod;
    private static volatile p0 getUploadFileMethod;
    private static volatile A0 serviceDescriptor;

    public static final class FileServiceBlockingStub extends b {
        public /* synthetic */ FileServiceBlockingStub(C0108g gVar, C0106f fVar, int i5) {
            this(gVar, fVar);
        }

        private FileServiceBlockingStub(C0108g gVar, C0106f fVar) {
            super(gVar, fVar);
        }

        public FileServiceBlockingStub build(C0108g gVar, C0106f fVar) {
            return new FileServiceBlockingStub(gVar, fVar);
        }
    }

    public static final class FileServiceFutureStub extends c {
        public /* synthetic */ FileServiceFutureStub(C0108g gVar, C0106f fVar, int i5) {
            this(gVar, fVar);
        }

        private FileServiceFutureStub(C0108g gVar, C0106f fVar) {
            super(gVar, fVar);
        }

        public FileServiceFutureStub build(C0108g gVar, C0106f fVar) {
            return new FileServiceFutureStub(gVar, fVar);
        }
    }

    public static abstract class FileServiceImplBase {
        public j batchThumbUpload(j jVar) {
            return B.f(FileServiceGrpc.getBatchThumbUploadMethod(), jVar);
        }

        public final z0 bindService() {
            s0 s0Var = new s0(FileServiceGrpc.getServiceDescriptor(), 0);
            p0 uploadFileMethod = FileServiceGrpc.getUploadFileMethod();
            q.J(uploadFileMethod, "method must not be null");
            s0Var.b(new y0(uploadFileMethod));
            p0 batchThumbUploadMethod = FileServiceGrpc.getBatchThumbUploadMethod();
            q.J(batchThumbUploadMethod, "method must not be null");
            s0Var.b(new y0(batchThumbUploadMethod));
            return s0Var.c();
        }

        public j uploadFile(j jVar) {
            return B.f(FileServiceGrpc.getUploadFileMethod(), jVar);
        }
    }

    public static final class FileServiceStub extends a {
        public /* synthetic */ FileServiceStub(C0108g gVar, C0106f fVar, int i5) {
            this(gVar, fVar);
        }

        /* JADX WARNING: type inference failed for: r4v1, types: [Y1.m0, java.lang.Object] */
        public j batchThumbUpload(j jVar) {
            C0114j h5 = getChannel().h(FileServiceGrpc.getBatchThumbUploadMethod(), getCallOptions());
            C0104e eVar = i.f6119a;
            f fVar = new f(h5);
            h5.start(new g(jVar, fVar), new Object());
            h5.request(2);
            return fVar;
        }

        /* JADX WARNING: type inference failed for: r4v1, types: [Y1.m0, java.lang.Object] */
        public j uploadFile(j jVar) {
            C0114j h5 = getChannel().h(FileServiceGrpc.getUploadFileMethod(), getCallOptions());
            C0104e eVar = i.f6119a;
            f fVar = new f(h5);
            h5.start(new g(jVar, fVar), new Object());
            h5.request(2);
            return fVar;
        }

        private FileServiceStub(C0108g gVar, C0106f fVar) {
            super(gVar, fVar);
        }

        public FileServiceStub build(C0108g gVar, C0106f fVar) {
            return new FileServiceStub(gVar, fVar);
        }
    }

    private FileServiceGrpc() {
    }

    public static p0 getBatchThumbUploadMethod() {
        p0 p0Var = getBatchThumbUploadMethod;
        if (p0Var == null) {
            synchronized (FileServiceGrpc.class) {
                try {
                    p0Var = getBatchThumbUploadMethod;
                    if (p0Var == null) {
                        o0 o0Var = o0.f2794E;
                        String a5 = p0.a("filecrud.FileService", "BatchThumbUpload");
                        BatchThumbUploadRequest defaultInstance = BatchThumbUploadRequest.getDefaultInstance();
                        C0385x xVar = C0453c.f5487a;
                        p0Var = new p0(o0Var, a5, new C0452b(defaultInstance), new C0452b(BatchThumbUploadResponse.getDefaultInstance()), true);
                        getBatchThumbUploadMethod = p0Var;
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return p0Var;
    }

    public static A0 getServiceDescriptor() {
        A0 a02 = serviceDescriptor;
        if (a02 == null) {
            synchronized (FileServiceGrpc.class) {
                try {
                    a02 = serviceDescriptor;
                    if (a02 == null) {
                        s0 s0Var = new s0("filecrud.FileService", 0);
                        p0 uploadFileMethod = getUploadFileMethod();
                        q.J(uploadFileMethod, "method");
                        ((List) s0Var.f5964F).add(uploadFileMethod);
                        p0 batchThumbUploadMethod = getBatchThumbUploadMethod();
                        q.J(batchThumbUploadMethod, "method");
                        ((List) s0Var.f5964F).add(batchThumbUploadMethod);
                        a02 = new A0(s0Var);
                        serviceDescriptor = a02;
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return a02;
    }

    public static p0 getUploadFileMethod() {
        p0 p0Var = getUploadFileMethod;
        if (p0Var == null) {
            synchronized (FileServiceGrpc.class) {
                try {
                    p0Var = getUploadFileMethod;
                    if (p0Var == null) {
                        o0 o0Var = o0.f2794E;
                        String a5 = p0.a("filecrud.FileService", "UploadFile");
                        FileUploadRequest defaultInstance = FileUploadRequest.getDefaultInstance();
                        C0385x xVar = C0453c.f5487a;
                        p0Var = new p0(o0Var, a5, new C0452b(defaultInstance), new C0452b(FileUploadResponse.getDefaultInstance()), true);
                        getUploadFileMethod = p0Var;
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return p0Var;
    }

    /* JADX WARNING: type inference failed for: r0v0, types: [io.grpc.stub.d, java.lang.Object] */
    public static FileServiceBlockingStub newBlockingStub(C0108g gVar) {
        return (FileServiceBlockingStub) b.newStub(new Object(), gVar);
    }

    /* JADX WARNING: type inference failed for: r0v0, types: [io.grpc.stub.d, java.lang.Object] */
    public static FileServiceFutureStub newFutureStub(C0108g gVar) {
        return (FileServiceFutureStub) c.newStub(new Object(), gVar);
    }

    /* JADX WARNING: type inference failed for: r0v0, types: [io.grpc.stub.d, java.lang.Object] */
    public static FileServiceStub newStub(C0108g gVar) {
        return (FileServiceStub) a.newStub(new Object(), gVar);
    }
}

File: app/src/main/java/com/hawkshaw/library/FileServiceGrpc.kt
package com.hawkshaw.library

import io.grpc.MethodDescriptor
import io.grpc.ServiceDescriptor
import io.grpc.Status
import io.grpc.stub.ClientCalls
import io.grpc.stub.ServerCalls
import io.grpc.stub.StreamObserver
import io.grpc.stub.AbstractStub
import io.grpc.stub.ServerCallStreamObserver
import io.grpc.CallOptions
import io.grpc.Channel
import io.grpc.stub.AbstractAsyncStub
import io.grpc.stub.AbstractBlockingStub
import io.grpc.stub.AbstractFutureStub
import java.util.concurrent.TimeUnit
import com.google.common.util.concurrent.ListenableFuture
import java.io.InputStream
import android.util.Log // Added for logging

/**
 * Service for file uploads
 */
class FileServiceGrpc private constructor() {
    companion object {
        private const val TAG = "FileServiceGrpcLogs" // Added for logging
        const val SERVICE_NAME = "filecrud.FileService"
        
        private const val METHODID_UPLOAD_FILE = 0
        private const val METHODID_BATCH_THUMB_UPLOAD = 1
        
        @JvmStatic
        private var serviceDescriptor: ServiceDescriptor? = null
        
        @JvmStatic
        private var uploadFileMethod: MethodDescriptor<FileUploadRequest, FileUploadResponse>? = null
        
        @JvmStatic
        private var batchThumbUploadMethod: MethodDescriptor<BatchThumbUploadRequest, BatchThumbUploadResponse>? = null
        
        /**
         * Creates a new blocking stub for the FileService service
         */
        @JvmStatic
        fun newBlockingStub(channel: Channel): FileServiceBlockingStub {
            Log.d(TAG, "[DEBUG] newBlockingStub called. Channel: $channel")
            val stub = FileServiceBlockingStub(channel)
            Log.d(TAG, "[DEBUG] newBlockingStub created: $stub")
            return stub
        }
        
        /**
         * Creates a new async stub for the FileService service
         */
        @JvmStatic
        fun newStub(channel: Channel): FileServiceStub {
            Log.d(TAG, "[DEBUG] newStub called. Channel: $channel")
            val stub = FileServiceStub(channel)
            Log.d(TAG, "[DEBUG] newStub created: $stub")
            return stub
        }
        
        /**
         * Creates a new ListenableFuture-style stub for the FileService service
         */
        @JvmStatic
        fun newFutureStub(channel: Channel): FileServiceFutureStub {
            Log.d(TAG, "[DEBUG] newFutureStub called. Channel: $channel")
            val stub = FileServiceFutureStub(channel)
            Log.d(TAG, "[DEBUG] newFutureStub created: $stub")
            return stub
        }
        
        /**
         * Gets the UploadFile method descriptor
         */
        @JvmStatic
        fun getUploadFileMethod(): MethodDescriptor<FileUploadRequest, FileUploadResponse> {
            Log.d(TAG, "[DEBUG] getUploadFileMethod called.")
            val method = uploadFileMethod
            if (method != null) {
                Log.d(TAG, "[DEBUG] getUploadFileMethod: Returning cached method descriptor.")
                return method
            }
            
            synchronized(FileServiceGrpc::class.java) {
                Log.d(TAG, "[DEBUG] getUploadFileMethod: Entered synchronized block.")
                val cached = uploadFileMethod
                if (cached != null) {
                    Log.d(TAG, "[DEBUG] getUploadFileMethod: Returning cached method descriptor from synchronized block.")
                    return cached
                }
                
                Log.d(TAG, "[DEBUG] getUploadFileMethod: Creating new method descriptor for UploadFile.")
                val fullMethodName = generateFullMethodName(SERVICE_NAME, "UploadFile")
                Log.d(TAG, "[DEBUG] getUploadFileMethod: Full method name: \"$fullMethodName\"")

                val methodBuilder = MethodDescriptor.newBuilder<FileUploadRequest, FileUploadResponse>()
                    .setType(MethodDescriptor.MethodType.BIDI_STREAMING)
                    .setFullMethodName(fullMethodName)
                    .setSampledToLocalTracing(true)
                
                // Create custom marshallers for non-protobuf objects
                val requestMarshaller = object : MethodDescriptor.Marshaller<FileUploadRequest> {
                    override fun stream(value: FileUploadRequest): InputStream {
                        Log.d(TAG, "[DEBUG] UploadFileRequestMarshaller: stream called with value: $value. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                    
                    override fun parse(stream: InputStream): FileUploadRequest {
                        Log.d(TAG, "[DEBUG] UploadFileRequestMarshaller: parse called with stream: $stream. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                }
                
                val responseMarshaller = object : MethodDescriptor.Marshaller<FileUploadResponse> {
                    override fun stream(value: FileUploadResponse): InputStream {
                        Log.d(TAG, "[DEBUG] UploadFileResponseMarshaller: stream called with value: $value. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                    
                    override fun parse(stream: InputStream): FileUploadResponse {
                        Log.d(TAG, "[DEBUG] UploadFileResponseMarshaller: parse called with stream: $stream. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                }
                Log.d(TAG, "[DEBUG] getUploadFileMethod: Setting custom request and response marshallers.")
                methodBuilder
                    .setRequestMarshaller(requestMarshaller)
                    .setResponseMarshaller(responseMarshaller)
                
                val methodDescriptor = methodBuilder.build()
                uploadFileMethod = methodDescriptor
                Log.d(TAG, "[DEBUG] getUploadFileMethod: New method descriptor created and cached: $methodDescriptor")
                return methodDescriptor
            }
        }
        
        /**
         * Gets the BatchThumbUpload method descriptor
         */
        @JvmStatic
        fun getBatchThumbUploadMethod(): MethodDescriptor<BatchThumbUploadRequest, BatchThumbUploadResponse> {
            Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod called.")
            val method = batchThumbUploadMethod
            if (method != null) {
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Returning cached method descriptor.")
                return method
            }
            
            synchronized(FileServiceGrpc::class.java) {
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Entered synchronized block.")
                val cached = batchThumbUploadMethod
                if (cached != null) {
                    Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Returning cached method descriptor from synchronized block.")
                    return cached
                }
                
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Creating new method descriptor for BatchThumbUpload.")
                val fullMethodName = generateFullMethodName(SERVICE_NAME, "BatchThumbUpload")
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Full method name: \"$fullMethodName\"")

                val methodBuilder = MethodDescriptor.newBuilder<BatchThumbUploadRequest, BatchThumbUploadResponse>()
                    .setType(MethodDescriptor.MethodType.BIDI_STREAMING)
                    .setFullMethodName(fullMethodName)
                    .setSampledToLocalTracing(true)

                val requestMarshaller = object : MethodDescriptor.Marshaller<BatchThumbUploadRequest> {
                    override fun stream(value: BatchThumbUploadRequest): InputStream {
                        Log.d(TAG, "[DEBUG] BatchThumbUploadRequestMarshaller: stream called with value: $value. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                    
                    override fun parse(stream: InputStream): BatchThumbUploadRequest {
                        Log.d(TAG, "[DEBUG] BatchThumbUploadRequestMarshaller: parse called with stream: $stream. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                }
                
                val responseMarshaller = object : MethodDescriptor.Marshaller<BatchThumbUploadResponse> {
                    override fun stream(value: BatchThumbUploadResponse): InputStream {
                        Log.d(TAG, "[DEBUG] BatchThumbUploadResponseMarshaller: stream called with value: $value. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                    
                    override fun parse(stream: InputStream): BatchThumbUploadResponse {
                        Log.d(TAG, "[DEBUG] BatchThumbUploadResponseMarshaller: parse called with stream: $stream. Throwing UnsupportedOperationException.")
                        throw UnsupportedOperationException("Not implemented")
                    }
                }
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: Setting custom request and response marshallers.")
                methodBuilder
                    .setRequestMarshaller(requestMarshaller)
                    .setResponseMarshaller(responseMarshaller)
                
                val methodDescriptor = methodBuilder.build()
                batchThumbUploadMethod = methodDescriptor
                Log.d(TAG, "[DEBUG] getBatchThumbUploadMethod: New method descriptor created and cached: $methodDescriptor")
                return methodDescriptor
            }
        }
        
        /**
         * Gets the service descriptor
         */
        @JvmStatic
        fun getServiceDescriptor(): ServiceDescriptor {
            Log.d(TAG, "[DEBUG] getServiceDescriptor called.")
            val descriptor = serviceDescriptor
            if (descriptor != null) {
                Log.d(TAG, "[DEBUG] getServiceDescriptor: Returning cached service descriptor.")
                return descriptor
            }
            
            synchronized(FileServiceGrpc::class.java) {
                Log.d(TAG, "[DEBUG] getServiceDescriptor: Entered synchronized block.")
                val cached = serviceDescriptor
                if (cached != null) {
                    Log.d(TAG, "[DEBUG] getServiceDescriptor: Returning cached service descriptor from synchronized block.")
                    return cached
                }
                
                Log.d(TAG, "[DEBUG] getServiceDescriptor: Creating new service descriptor. ServiceName: \"$SERVICE_NAME\"")
                val result = ServiceDescriptor.newBuilder(SERVICE_NAME)
                    .addMethod(getUploadFileMethod()) // getUploadFileMethod itself logs
                    .addMethod(getBatchThumbUploadMethod()) // getBatchThumbUploadMethod itself logs
                    .build()
                
                serviceDescriptor = result
                Log.d(TAG, "[DEBUG] getServiceDescriptor: New service descriptor created and cached: $result")
                return result
            }
        }
        
        private fun generateFullMethodName(serviceName: String, methodName: String): String {
            val fullName = "$serviceName/$methodName"
            Log.d(TAG, "[DEBUG] generateFullMethodName called. ServiceName: \"$serviceName\", MethodName: \"$methodName\", Result: \"$fullName\"")
            return fullName
        }
    }
    
    /**
     * A stub to allow clients to do synchronous calls to the FileService service
     */
    class FileServiceBlockingStub @JvmOverloads constructor(
        channel: Channel,
        callOptions: CallOptions = CallOptions.DEFAULT
    ) : AbstractBlockingStub<FileServiceBlockingStub>(channel, callOptions) {
        init {
            Log.d(TAG, "[DEBUG] FileServiceBlockingStub initialized. Channel: $channel, CallOptions: $callOptions")
        }
        override fun build(channel: Channel, callOptions: CallOptions): FileServiceBlockingStub {
            Log.d(TAG, "[DEBUG] FileServiceBlockingStub.build called. Channel: $channel, CallOptions: $callOptions")
            val newStub = FileServiceBlockingStub(channel, callOptions)
            Log.d(TAG, "[DEBUG] FileServiceBlockingStub.build: New stub created: $newStub")
            return newStub
        }
    }
    
    /**
     * A stub to allow clients to do ListenableFuture-style calls to the FileService service
     */
    class FileServiceFutureStub @JvmOverloads constructor(
        channel: Channel,
        callOptions: CallOptions = CallOptions.DEFAULT
    ) : AbstractFutureStub<FileServiceFutureStub>(channel, callOptions) {
        init {
            Log.d(TAG, "[DEBUG] FileServiceFutureStub initialized. Channel: $channel, CallOptions: $callOptions")
        }
        override fun build(channel: Channel, callOptions: CallOptions): FileServiceFutureStub {
            Log.d(TAG, "[DEBUG] FileServiceFutureStub.build called. Channel: $channel, CallOptions: $callOptions")
            val newStub = FileServiceFutureStub(channel, callOptions)
            Log.d(TAG, "[DEBUG] FileServiceFutureStub.build: New stub created: $newStub")
            return newStub
        }
    }
    
    /**
     * A stub to allow clients to do asynchronous calls to the FileService service
     */
    class FileServiceStub @JvmOverloads constructor(
        channel: Channel,
        callOptions: CallOptions = CallOptions.DEFAULT
    ) : AbstractAsyncStub<FileServiceStub>(channel, callOptions) {
        init {
            Log.d(TAG, "[DEBUG] FileServiceStub initialized. Channel: $channel, CallOptions: $callOptions")
        }
        /**
         * Upload a file with streaming
         */
        fun uploadFile(responseObserver: StreamObserver<FileUploadResponse>): StreamObserver<FileUploadRequest> {
            Log.d(TAG, "[DEBUG] FileServiceStub.uploadFile called. ResponseObserver: $responseObserver, CallOptions: $callOptions")
            val call = ClientCalls.asyncBidiStreamingCall(
                channel.newCall(getUploadFileMethod(), callOptions), // getUploadFileMethod itself logs
                responseObserver
            )
            Log.d(TAG, "[DEBUG] FileServiceStub.uploadFile: ClientCalls.asyncBidiStreamingCall invoked, returning StreamObserver: $call")
            return call
        }
        
        /**
         * Upload multiple thumbnails in batch
         */
        fun batchThumbUpload(responseObserver: StreamObserver<BatchThumbUploadResponse>): StreamObserver<BatchThumbUploadRequest> {
            Log.d(TAG, "[DEBUG] FileServiceStub.batchThumbUpload called. ResponseObserver: $responseObserver, CallOptions: $callOptions")
            val call = ClientCalls.asyncBidiStreamingCall(
                channel.newCall(getBatchThumbUploadMethod(), callOptions), // getBatchThumbUploadMethod itself logs
                responseObserver
            )
            Log.d(TAG, "[DEBUG] FileServiceStub.batchThumbUpload: ClientCalls.asyncBidiStreamingCall invoked, returning StreamObserver: $call")
            return call
        }
        
        override fun build(channel: Channel, callOptions: CallOptions): FileServiceStub {
            Log.d(TAG, "[DEBUG] FileServiceStub.build called. Channel: $channel, CallOptions: $callOptions")
            val newStub = FileServiceStub(channel, callOptions)
            Log.d(TAG, "[DEBUG] FileServiceStub.build: New stub created: $newStub")
            return newStub
        }
    }
    
    /**
     * Base class for server implementations of the FileService service
     */
    abstract class FileServiceImplBase {
        init {
            Log.d(TAG, "[DEBUG] FileServiceImplBase initialized.")
        }
        /**
         * Upload a file with streaming
         */
        open fun uploadFile(responseObserver: StreamObserver<FileUploadResponse>): StreamObserver<FileUploadRequest> {
            Log.d(TAG, "[DEBUG] FileServiceImplBase.uploadFile called. ResponseObserver: $responseObserver. Returning asyncUnimplementedStreamingCall.")
            return ServerCalls.asyncUnimplementedStreamingCall(getUploadFileMethod(), responseObserver) // getUploadFileMethod itself logs
        }
        
        /**
         * Upload multiple thumbnails in batch
         */
        open fun batchThumbUpload(responseObserver: StreamObserver<BatchThumbUploadResponse>): StreamObserver<BatchThumbUploadRequest> {
            Log.d(TAG, "[DEBUG] FileServiceImplBase.batchThumbUpload called. ResponseObserver: $responseObserver. Returning asyncUnimplementedStreamingCall.")
            return ServerCalls.asyncUnimplementedStreamingCall(getBatchThumbUploadMethod(), responseObserver) // getBatchThumbUploadMethod itself logs
        }
        
        /**
         * Bind service methods to their handlers
         */
        fun bindService(): io.grpc.ServerServiceDefinition {
            Log.d(TAG, "[DEBUG] FileServiceImplBase.bindService called. ServiceName: \"$SERVICE_NAME\"")
            val builder = io.grpc.ServerServiceDefinition.builder(SERVICE_NAME)
                .addMethod(
                    getUploadFileMethod(), // getUploadFileMethod itself logs
                    ServerCalls.asyncBidiStreamingCall { innerResponseObserver ->
                        Log.d(TAG, "[DEBUG] FileServiceImplBase.bindService: UploadFile handler invoked. Calling outer uploadFile.")
                        uploadFile(innerResponseObserver) // This logs inside uploadFile
                    }
                )
                .addMethod(
                    getBatchThumbUploadMethod(), // getBatchThumbUploadMethod itself logs
                    ServerCalls.asyncBidiStreamingCall { innerResponseObserver ->
                        Log.d(TAG, "[DEBUG] FileServiceImplBase.bindService: BatchThumbUpload handler invoked. Calling outer batchThumbUpload.")
                        batchThumbUpload(innerResponseObserver) // This logs inside batchThumbUpload
                    }
                )
            val serviceDef = builder.build()
            Log.d(TAG, "[DEBUG] FileServiceImplBase.bindService: ServerServiceDefinition built: $serviceDef")
            return serviceDef
        }
    }
}
